src/analysis/retail/priest/discipline/modules/spells/PowerWordShield.tsx

```diff
@@ -128,34 +131,51 @@ class PowerWordShield extends Analyzer {
 
   handleRemovedShield(event: RefreshBuffEvent | RemoveBuffEvent) {
     const info = this.shieldApplications.get(event.targetID);
+
     if (
       !info ||
-      info.event.timestamp > event.timestamp || // not sure how this happens? fabrication stuff?
+      info.event.timestamp > event.timestamp ||
       info.event.timestamp + POWER_WORD_SHIELD_DURATION_MS < event.timestamp
     ) {
       return;
     }
-
-    const shieldAmount = info.event.absorb || 0; // the initial amount, from the ApplyBuffEvent
-    const shieldOfAbsolutionBonus = info.crit
+    const shieldAmount = info.event.absorb || 0; // the initial amount, from the ApplyBuffEvent/RefreshBuffEvent
+    let shieldOfAbsolutionBonus = info.crit
       ? info.shieldOfAbsolutionValue * 2
       : info.shieldOfAbsolutionValue; // the amount the shield was increased by the 4p
-    const basePowerWordShieldAmount = (shieldAmount - shieldOfAbsolutionBonus) / 1.5;
+    shieldOfAbsolutionBonus *= info.rapture ? 1.3 : 1;
+    const basePowerWordShieldAmount = this.hasAegis
+      ? (shieldAmount - shieldOfAbsolutionBonus) / 1.5
+      : shieldAmount - shieldOfAbsolutionBonus;
     let totalShielded = info.healing; // this is the amount of healing the shield did
+    this.pwsValue +=
+      totalShielded - basePowerWordShieldAmount > 0 ? basePowerWordShieldAmount : totalShielded;
 
-    this.pwsValue += basePowerWordShieldAmount;
-    totalShielded -= basePowerWordShieldAmount;
-    this.t23pValue +=
-      totalShielded >= shieldOfAbsolutionBonus ? shieldOfAbsolutionBonus : totalShielded;
+    totalShielded -=
+      totalShielded - basePowerWordShieldAmount > 0 ? basePowerWordShieldAmount : totalShielded;
```

Comment:
this repeated value (`basePowerWordShieldAmount > 0 ? basePowerWordShieldAmount : totalShielded`) should get a variable name